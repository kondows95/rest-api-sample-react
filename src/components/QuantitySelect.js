import React from 'react';
import PropTypes from 'prop-types'
import NativeSelect from '@material-ui/core/NativeSelect';

const QuantitySelect = ({ id, quantity, maxQuantity, changeQuantity }) => {
  const options = []
  for (let i = 1; i <= Math.max(20, maxQuantity); i++) {
    options.push(
      <option key={i} value={i}>{i}</option>
    )
  }
  return (
    <NativeSelect
      value={quantity}
      onChange={(event) => changeQuantity(id, event.target.value)}
    >
      {options}
    </NativeSelect>
  )
}

QuantitySelect.propTypes = {
  id: PropTypes.number.isRequired,
  quantity: PropTypes.number.isRequired,
  maxQuantity: PropTypes.number.isRequired,
  changeQuantity: PropTypes.func,
}

export default QuantitySelect
