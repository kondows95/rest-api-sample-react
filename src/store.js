import rootReducer from './modules/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
export default createStore(rootReducer, applyMiddleware(thunk));