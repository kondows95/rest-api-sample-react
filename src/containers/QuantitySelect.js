import { connect } from 'react-redux'
import QuantitySelect from '../components/QuantitySelect'
import { changeQuantity } from '../modules/cart'

export default connect(
  (state) => ({
    maxQuantity: state.cart.maxQuantity
  }),
  (dispatch) => ({
    changeQuantity: (itemId, quantity) => dispatch(changeQuantity(itemId, quantity)),    
  })
)(QuantitySelect)
