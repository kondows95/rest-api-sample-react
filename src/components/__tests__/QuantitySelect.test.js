import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import store from '../../store';
import QuantitySelect from '../QuantitySelect';
let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})
afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})
const changeQuantity = jest.fn((id, quantity) => {
  let quantity1 = quantity;
});
const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
describe("QuantitySelect component", () => {
  it('matches the snapshot', () => {
    const QuantitySelectSnapshot = renderer.create(
      <QuantitySelect
        id={1}
        quantity={2}
        maxQuantity={4}
        changeQuantity={changeQuantity} />
    ).toJSON();
    expect(QuantitySelectSnapshot).toMatchSnapshot();
  });
});

describe("testing Quantity Select", () => {
  it('testing changing quantity', () => {
    const options =[1,2,3,4];
    act(() => {
      ReactDOM.render((
        <QuantitySelect
        id={1}
        quantity={3}
        maxQuantity={4}
        changeQuantity={changeQuantity} />
      ), container);
    });
    const selectArr = document.querySelectorAll('select');
    act(() =>{
      selectArr[0].value = "6";
     selectArr[0].dispatchEvent(new Event('change', {bubbles: true}))
    })
    expect(changeQuantity).toHaveBeenCalledWith(1,"6");
  });
 
})