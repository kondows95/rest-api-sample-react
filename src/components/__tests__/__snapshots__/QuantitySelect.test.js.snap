// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantitySelect component matches the snapshot 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  onClick={[Function]}
>
  <select
    className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    value={2}
  >
    <option
      value={1}
    >
      1
    </option>
    <option
      value={2}
    >
      2
    </option>
    <option
      value={3}
    >
      3
    </option>
    <option
      value={4}
    >
      4
    </option>
    <option
      value={5}
    >
      5
    </option>
    <option
      value={6}
    >
      6
    </option>
    <option
      value={7}
    >
      7
    </option>
    <option
      value={8}
    >
      8
    </option>
    <option
      value={9}
    >
      9
    </option>
    <option
      value={10}
    >
      10
    </option>
    <option
      value={11}
    >
      11
    </option>
    <option
      value={12}
    >
      12
    </option>
    <option
      value={13}
    >
      13
    </option>
    <option
      value={14}
    >
      14
    </option>
    <option
      value={15}
    >
      15
    </option>
    <option
      value={16}
    >
      16
    </option>
    <option
      value={17}
    >
      17
    </option>
    <option
      value={18}
    >
      18
    </option>
    <option
      value={19}
    >
      19
    </option>
    <option
      value={20}
    >
      20
    </option>
  </select>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root MuiNativeSelect-icon"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;
